src/components/shared/useUserEventHandlers.ts

```diff
@@ -29,9 +35,38 @@ export const useUserEventHandlers = ({
   const createEvent = useElementEvent(element);
 
   const transformation = useTransform(transform);
+  const prevBinInfoRef = useRef<BinInfo | undefined>(element.binInfo);
+  const prevValueRef = useRef<string>('');
+  const prevSelectedNetworkRef = useRef<CardBrand | undefined>(element.selectedNetwork);
+
+
+  useEffect(() => {
+    const currentBinInfo = element.binInfo;
+    const prevBinInfo = prevBinInfoRef.current;
+
+    if (currentBinInfo !== prevBinInfo && onChange) {
+      const event = createEvent(prevValueRef.current);
+      onChange(event);
+    }
+
+    prevBinInfoRef.current = currentBinInfo;
+  }, [element.binInfo, onChange, createEvent, element.id]);
+
+  useEffect(() => {
+    const currentSelectedNetwork = element.selectedNetwork;
+    const prevSelectedNetwork = prevSelectedNetworkRef.current;
+
+    if (currentSelectedNetwork !== prevSelectedNetwork && onChange) {
+      const event = createEvent(prevValueRef.current);
+      onChange(event);
+    }
+
+    prevSelectedNetworkRef.current = currentSelectedNetwork;
+  }, [element.selectedNetwork, onChange, createEvent, element.id]);
```

Comment:
We're storing metadata in `useRef` but input values in `_elementValues`. This means if the component unmounts, we lose the BIN info but keep the typed value.

Should probably store `binInfo` and `selectedNetwork` in the global state too:

```typescript
// ElementValues.ts
const _elementMetadata: Record<string, {
  binInfo?: BinInfo;
  selectedNetwork?: CardBrand;
}> = {};

// Then compare against previous values from global state instead of refs
const prevBinInfo = _elementMetadata[element.id]?.binInfo;
if (currentBinInfo !== prevBinInfo && onChange) {
  // fire event
}
_elementMetadata[element.id] = { ...prev, binInfo: currentBinInfo };
```

This way everything stays consistent and we don't lose metadata on remounts.